### ISSUE TITLE
Unread count for channel 

### ISSUE THREAD

**Issue by maxmaxandr** [GetStream/stream-chat-android#475]
I subscribe on user's channels & want to get unreadCount in the channel. 
For the first time, I got the right result of unreadCount in the channel but the next time when for example somebody sends me a message unreadCount is not changed and have a wrong value.

```
     val queryChannelResult = chatDomain.useCases.queryChannels(filter, null).execute()
            val queryChannelsController = queryChannelResult.data()
            queryChannelsController.channels.observeForever {
                for (channel in it) {
//check channel.unreadCount 
                }
            }
```
Also if I got a event NewMessageEvent,  it has a message with the channel but channel.unreadCount is always null.

**Comment by tschellenbach** [comment]
That's weird since the count is computed based on livedata objects such as messages, read etc. Going to investigate

**Comment by tschellenbach** [comment]
@maxmaxandr message has a channel object but that's not the full channel object used in other parts of the library. @elevenetc is reworking our type system to better reflect that (actually just having the channel id on the message is probably adequate.)

I don't get why channel.unreadCount is out of sync for you though. The unread count is recomputed whenever the channel changes. (computeUnreadCount(domainImpl.currentUser, _read.value, messages)

I'm trying to build a test case that replicates this. Are you sure it's a new message that doesn't increment the count? Or is there a different scenario to test?

**Comment by maxmaxandr** [comment]
> @maxmaxandr message has a channel object but that's not the full channel object used in other parts of the library. @elevenetc is reworking our type system to better reflect that (actually just having the channel id on the message is probably adequate.)
> 
> I don't get why channel.unreadCount is out of sync for you though. The unread count is recomputed whenever the channel changes. (computeUnreadCount(domainImpl.currentUser, _read.value, messages)
> 
> I'm trying to build a test case that replicates this. Are you sure it's a new message that doesn't increment the count? Or is there a different scenario to test?

Yes, I'am checked a few times that after receive a new message unreadCount in channel observable doesn't changed.    I can send a 10 messages and counter will not changed - but it can be changed after 11 message. I will provide logs with it.

**Comment by tschellenbach** [comment]
Thanks. Also note that your own messages aren't counted.

**Comment by tschellenbach** [comment]
I created a failing test that replicates this issue:
https://github.com/GetStream/stream-chat-android-livedata/blob/475/livedata/src/test/java/io/getstream/chat/android/livedata/usecase/QueryChannelsImplTest.kt#L33

For unknown reasons the count on the channelController is updated, but the count exposed by the queryChannelsController is somehow not picking up the change. Will keep you posted.

**Comment by tschellenbach** [comment]
Fixed in the 0.4.5 release. Thanks for reporting this.

https://github.com/GetStream/stream-chat-android-livedata/releases/tag/0.4.5
